subprojects/core/src/integTest/groovy/org/gradle/api/NamedDomainObjectContainerIntegrationTest.groovy

```diff
@@ -16,15 +16,15 @@
 
 package org.gradle.api
 
+import groovy.transform.SelfType
 import spock.lang.Issue
 
-class NamedDomainObjectContainerIntegrationTest extends AbstractDomainObjectContainerIntegrationTest {
-    @Override
+@SelfType(AbstractDomainObjectContainerIntegrationTest)
+trait AbstractNamedDomainObjectContainerIntegrationTest {
```

Comment:
Interesting, I didn't know that a `trait` with a `setup()` method works with spock.

Reply:
According to the doc https://spockframework.org/spock/docs/2.0-M4/all_in_one.html#_using_traits_with_specifications 

> Traits on Specifications are not supported by Spock, some use-cases might work while others don’t. This is due to how groovy implements traits and AST transformations.

So we're lucky here to have it working. If it doesn't work in the future we can easily extract the setup method into a static method.

Reply:
What I have been doing is to use `@Before` on a method in a trait and that worked. Can you use that here as well?

Reply:
You mean JUnit4 `@Before`? It seems to be removed in latest spock: https://github.com/spockframework/spock/issues/1152